src/components/DataCollector/index.js

```diff
@@ -13,62 +13,57 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import CONFIG_DOC_URI from "../../constants/docUri";
+import EVENT_COMMAND_DOC_URI from "../../constants/docUri";
 
-const createDataCollector = ({ eventManager, logger }) => {
+const createDataCollector = ({ eventManager }) => {
   return {
     commands: {
-      event(options) {
-        const { errors, warnings } = validateUserEventOptions(options);
```

Comment:
I have to think about this PR a little more, but from a first glance, and for the sake of keeping Components commands' API simple, I'd rather we convert the `validation message` into some sort of a template / util, and each component will use it to throw its errors if any, instead of changing the API of the commands and making it more verbose.

I'm open to discussing it though.

Reply:
If a component throws the error it is captured and presented as an error. The component command does not need to return errors and warnings. See the Personalization component as an example.